### PR TITLE
Make deploy and deploy-dev rule of Makefile work

### PR DESCRIPTION
**Pull request by hougangliu**
When I run `make deploy` and `make deploy-dev`, deploying always fails. The main root cause is that the kustomize directory layout of `config/default` is not right for [the issue](https://github.com/kubernetes-sigs/kubebuilder/issues/595)

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/95)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,29 +18,26 @@ run: generate fmt vet
...
-undeploy-dev: manifests
+undeploy-dev:
 	kustomize build config/overlays/development | kubectl delete -f -
-	kubectl delete -f config/crds
 
 # Generate manifests e.g. CRD, RBAC etc.
 manifests:
-	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
+	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd --output-dir=config/default/crds
+	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go rbac --output-dir=config/default/rbac
```
Comment (yuzisun) [inline Makefile]: Do we also need to generate manager manifests ? Any reason why you do not want to do all ?

**Comment by yuzisun** [comment]
Thanks for fixing this. Does this still work with old version of kustomize? I assume the issue is with 2.0.

File: Makefile (hunk 1 above)
Comment (hougangliu) [reply Makefile]: `go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all` only generate `crds` and `brace`

**Comment by hougangliu** [comment]
> kustomize

I tested my fix both with  kustomize_1.0.10 and kustomize_2.0.3, everything goes well

**Comment by yuzisun** [comment]
double checked, this does look better organized! 
/lgtm
/approve

**Pull request merged**